### PR TITLE
test(ENG-75): add missing interest math test vectors per SPEC §8.1

### PR DESCRIPTION
**Pull request by Connorbelez**
## Summary
- Add month boundary crossing test: Jan 28 → Feb 3 = 7 days
- Add 30-day accrual test: 10%/100%/$100K/30d = $821.92
- Add 1-day high-rate test: 12%/25%/$200K/1d = $16.44
- Add precision tests verifying no premature rounding in intermediate calculations

## Test plan
- [x] All 45 tests pass (`bun test convex/accrual/__tests__/interestMath.test.ts`)
- [x] `bun check` passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add additional interest accrual test coverage, including date range edge cases and floating-point precision guarantees.

Tests:
- Add daysBetween test for month boundary spans.
- Add calculatePeriodAccrual tests for specific rates, ownership percentages, principals, and day counts to validate expected dollar outputs.
- Add precision-focused tests to ensure no premature rounding and consistent results when summing multiple periods.

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage for interest calculation functions to ensure accurate day counting and accrual computations across various scenarios and date boundaries.
  * Added precision validation tests to verify calculations maintain accuracy without premature rounding errors, ensuring consistent results across different calculation methods.

### CODE REVIEW

File: convex/accrual/__tests__/interestMath.test.ts [hunk 1]
```diff
@@ -182,6 +187,18 @@ describe("calculatePeriodAccrual", () => {
...
 	});
+
+	it("computes $821.92 for 10% rate, 100% ownership, $100K, 30 days", () => {
+		// (0.10 * 1.0 * 100_000 * 30) / 365 = 821.9178... → rounds to $821.92
+		const result = calculatePeriodAccrual(0.1, 1, 100_000, 30);
+		expect(result).toBeCloseTo(821.92, 2);
+	});
+
+	it("computes $16.44 for 12% rate, 25% ownership, $200K, 1 day", () => {
+		// (0.12 * 0.25 * 200_000 * 1) / 365 = 16.4383... → rounds to $16.44
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/accrual/__tests__/interestMath.test.ts]: Fixed: Updated comment to say approximately 16.44 instead of rounds to, since calculatePeriodAccrual returns the raw float without rounding.
